tests/integration/util/test_file_converter.py

```diff
@@ -81,3 +94,14 @@ def test_convert_netcdf_to_csv(netcdf_input_path, csv_filename):
             file.read()
     except FileNotFoundError as fnf_error:
         raise fnf_error
+
+
+def test_convert_csv_to_netcdf(csv_input_path, netcdf_filename):
+    """Test to check end-to-end processing of csv files and their
+     conversion into netcdf files."""
+    fc.convert_csv(csv_input_path, netcdf_filename)
+    try:
+        netCDF4.Dataset(netcdf_filename)
+    except FileNotFoundError as fnf_error:
+        raise fnf_error
```

Comment:
this test doesn't appear to be validating anything? If the test is intended to pass so long as no exceptions are thrown, then I would be inclined to make that a little more explicit:
```suggestion
    try:
        netCDF4.Dataset(netcdf_filename)
    except Exception as exc:
        pytest.fail(f"Unexpected exception: {e}")
```

Separately, there's no need to catch and re-raise like this. This code essentially duplicates the behaviour for when there's no try...catch, so can be omitted. In the suggestion above, the try...catch is doing something other than "bubbling up" the exception, it's handling it and marking the test as a failure which is why I haven't removed it

Reply:
This is interesting. I very much like your suggestion, but how does it actually handle the failure (and btw, you may have to explain using short words as I am just dipping in here to see if I can understand anything again yet)?

Reply:
Wait, I see now. I like it. I will certainly use your suggestion (as soon as I'm up to it). Thanks for this!

Reply:
I do have a few questions about this one.  Well, just one question really.  In my version, would this not actually fail upon a failure?

What I mean is; your version above explicitly raises a test fail, which I like because it's very clear in the code, but would it specify the failure as a FileNotFound error in the results? And would my version not actually produce a failure?

Reply:
What I think I might try is an amalgamation of your suggestion and the existing code; I would like to keep the FileNotFound error as this makes the results more clear, but I want to include the pytest.fail clarity in the code, so I might that now...